business/apps.go

```diff
@@ -63,19 +210,63 @@ func (in *AppService) GetAppList(namespace string) (models.AppList, error) {
 			Name:         keyApp,
 			IstioSidecar: true,
 		}
-		labels := make(map[string][]string)
+		applabels := make(map[string][]string)
+		apVsNames := make([]string, 0)
+		apDrNames := make([]string, 0)
 		for _, srv := range valueApp.Services {
-			joinMap(labels, srv.Labels)
+			joinMap(applabels, srv.Labels)
+
+			svcVirtualServices := kubernetes.FilterVirtualServices(virtualServices, srv.Namespace, srv.Name)
+			apVsNames = append(apVsNames, getIstioResourcesNames(svcVirtualServices)...)
+
+			svcDestinationRules := kubernetes.FilterDestinationRules(destinationRules, srv.Namespace, srv.Name)
+			apDrNames = append(apDrNames, getIstioResourcesNames(svcDestinationRules)...)
 		}
+		appItem.VirtualServices = apVsNames
+		appItem.DestinationRules = apDrNames
+
+		apGwNames := make([]string, 0)
+		apApNames := make([]string, 0)
+		apPaNames := make([]string, 0)
+		apScNames := make([]string, 0)
+		apRaNames := make([]string, 0)
+		apEfNames := make([]string, 0)
 		for _, wrk := range valueApp.Workloads {
-			joinMap(labels, wrk.Labels)
+			joinMap(applabels, wrk.Labels)
+
+			wSelector := labels.Set(wrk.Labels).AsSelector().String()
+			wGw := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, gateways)
+			apGwNames = append(apGwNames, getIstioResourcesNames(wGw)...)
+
+			wAp := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, authorizationPolicies)
+			apApNames = append(apApNames, getIstioResourcesNames(wAp)...)
+
+			wPa := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, peerAuthentications)
+			apPaNames = append(apPaNames, getIstioResourcesNames(wPa)...)
+
+			wSc := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, sidecars)
+			apScNames = append(apScNames, getIstioResourcesNames(wSc)...)
+
+			wRa := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, requestAuthentications)
+			apRaNames = append(apRaNames, getIstioResourcesNames(wRa)...)
+
+			wEf := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, envoyFilters)
+			apEfNames = append(apEfNames, getIstioResourcesNames(wEf)...)
 		}
-		appItem.Labels = buildFinalLabels(labels)
+		appItem.Labels = buildFinalLabels(applabels)
+		appItem.Gateways = apGwNames
+		appItem.AuthorizationPolicies = apApNames
+		appItem.PeerAuthentications = apPaNames
+		appItem.Sidecars = apScNames
+		appItem.RequestAuthentications = apRaNames
+		appItem.EnvoyFilters = apEfNames
```

Comment:
Another thing I see, just a proposal, is that you can get rid of some repetition and initialization by using an associative/struct array. The proposal feels more (a bit) closed to modification.

```go
	for _, wrk := range valueApp.Workloads {
		joinMap(applabels, wrk.Labels)

		wSelector := labels.Set(wrk.Labels).AsSelector().String()

		data := []struct {
			source []kubernetes.IstioObject
			destination *[]string
		} {
			{ source: gateways, destination: &appItem.Gateways },
			{ source: authorizationPolicies, destination: &appItem.AuthorizationPolicies },
			{ source: peerAuthentications, destination: &appItem.PeerAuthentications },
			{ source: sidecars, destination: &appItem.Sidecars },
			{ source: requestAuthentications, destination: &appItem.RequestAuthentications },
			{ source: envoyFilters, destination: &appItem.EnvoyFilters },
		}

		for _, pair := range data {
			wEf := kubernetes.FilterIstioObjectsForWorkloadSelector(wSelector, pair.source)
			*(pair.destination) = append(*(pair.destination), getIstioResourcesNames(wEf)...)
		}
	}
	appItem.Labels = buildFinalLabels(applabels)
```

Reply:
+1